model/src/main/proto/exploration.proto

```diff
@@ -0,0 +1,145 @@
+syntax = "proto3";
+
+import "subtitled_html.proto";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Data for a single exploration.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeExploration.kt
+message Exploration {
+  repeated State states = 1;
+  repeated ParamChange param_changes = 2;
+  repeated ParamSpec param_specs = 3;
+  string initStateName = 4;
+  string objective = 5;
+  bool correctness_feedback_enabled = 6;
+  string title = 7;
+  string language_code = 8;
+}
+
+// Data for a param change.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamChange.kt
+message ParamChange {
+  string generator_id = 1;
+  string name = 2;
+  CustomizationArgs customization_args = 3;
+}
+
+// Data for a param spec.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamSpec.kt
+message ParamSpec {
+  string obj_type = 1;
+}
+
+// Data for a single state
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeState.kt
+message State {
+  // Mapping from content_id to a VoiceoverMapping
+  map<string, VoiceoverMapping> recorded_voiceovers = 1;
+  SubtitledHtml content = 2;
+  // Mapping from content_id to a TranslationMapping
+  map<string, TranslationMapping>written_translations = 3;
+  repeated ParamChange param_changes = 4;
+  string classifier_model_id = 5;
+  Interaction interaction = 6;
+  bool solicit_answer_details = 7;
+}
+
+// Data for customization args.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeCustomizationArgs.kt
+message CustomizationArgs {
+  bool parse_with_jinja = 1;
+  string value = 2;
```

Comment:
Is value always type string? This was 'Any' in the GAE model.

Reply:
Based on a search of the codebase - https://github.com/oppia/oppia/search?l=YAML&p=2&q=customization_args
I could see that each of the customization args have a value field that is a string. Not sure if we need anything else apart from the value field? 
@seanlip is there a way to find out all the possible customization args we can get?

Reply:
Hi @vinitamurthi -- "customization args" is an overloaded term. They exist for interactions, parameters, and a whole bunch of other things.

For interactions: see extensions/interaction/NameOfInteraction and look at the python file. There's a list of customization arg descriptions in there, with their possible schemas.

For parameters, it's defined here: https://github.com/oppia/oppia/blob/develop/core/domain/param_domain.py#L99

Reply:
Separated customization args for param changes and interactions.
For interactions I noticed this comment: https://github.com/oppia/oppia/blob/1b668fd3c74f9481232a3c1c73ecd632036a902e/core/domain/state_domain.py#L340 where its a map from name to a default value.
Default value can be different types so I looked at the different possible values in extensions/interaction/NameOfInteraction and added them to an InteractionObject message